src/compiler/factory.ts

```diff
@@ -1262,19 +1170,30 @@ namespace ts {
         return reduceLeft(expressions, createComma);
     }
 
-    export function createExpressionFromEntityName(node: EntityName | Expression): Expression {
-        return isQualifiedName(node)
-            ? createPropertyAccess(
-                createExpressionFromEntityName(node.left),
-                getSynthesizedClone(node.right)
-            )
-            : getSynthesizedClone(node);
+    export function createExpressionFromEntityName(node: EntityName | Expression, emitOptions?: NodeEmitOptions): Expression {
+        if (isQualifiedName(node)) {
+            const left = createExpressionFromEntityName(node.left, emitOptions);
+            const right = getMutableClone(node.right, emitOptions && clone(emitOptions));
+            return createPropertyAccess(left, right, /*location*/ node, emitOptions && clone(emitOptions));
+        }
+        else if (isIdentifier(node)) {
+            return getMutableClone(node, emitOptions && clone(emitOptions));
+        }

```

Comment:
why have the `isIdentifier(node)` case? I think you are fine with just `else { getMutableClone( ...`